ui/pages/confirmations/components/confirm/info/hooks/use-token-values.ts

```diff
@@ -22,26 +22,42 @@ export const useTokenValues = (transactionMeta: TransactionMeta) => {
   const decodedResponse = useDecodedTransactionData();
   const { value, pending } = decodedResponse;
 
-  const decodedTransferValue = useMemo(() => {
-    if (!value || !decimals) {
-      return 0;
-    }
+  const { decodedTransferValue, isDecodedTransferValuePending } =
+    useMemo(() => {
+      if (!value) {
+        return {
+          decodedTransferValue: 0,
+          isDecodedTransferValuePending: false,
+        };
+      }
 
-    const paramIndex = value.data[0].params.findIndex(
-      (param) =>
-        param.value !== undefined &&
-        !isHexString(param.value) &&
-        param.value.length === undefined &&
-        !isBoolean(param.value),
-    );
-    if (paramIndex === -1) {
-      return 0;
-    }
+      if (!decimals) {
+        return { decodedTransferValue: 0, isDecodedTransferValuePending: true };
+      }
 
-    return new BigNumber(value.data[0].params[paramIndex].value.toString())
-      .dividedBy(new BigNumber(10).pow(Number(decimals)))
-      .toNumber();
-  }, [value, decimals]);
+      const paramIndex = value.data[0].params.findIndex(
+        (param) =>
+          param.value !== undefined &&
+          !isHexString(param.value) &&
+          param.value.length === undefined &&
+          !isBoolean(param.value),
+      );
+      if (paramIndex === -1) {
+        return {
+          decodedTransferValue: 0,
+          isDecodedTransferValuePending: false,
+        };
+      }
+
+      return {
+        decodedTransferValue: new BigNumber(
+          value.data[0].params[paramIndex].value.toString(),
+        )
+          .dividedBy(new BigNumber(10).pow(Number(decimals)))
+          .toNumber(),
```

Comment:
[nit] could use calcTokenAmount here. Then, we could return `toFixed` to ensure precision is preserved. Then there would be no need for `toNonScientificString` below

Reply:
Thanks for the comment. Let's address that after feedback from the team on a single PR (https://github.com/MetaMask/metamask-extension/pull/28212) instead of blocking both.